src/pheval/post_processing/post_processing.py

```diff
@@ -69,27 +75,51 @@ def as_dict(self):
         }
 
 
+@dataclass
+class PhEvalDiseaseResult(PhEvalResult):
+    """Minimal data required from tool-specific output for disease prioritisation."""
+
+    disease_name: str
+    disease_identifier: str
+    score: float
+
+
+@dataclass
+class RankedPhEvalDiseaseResult(PhEvalResult):
+    """PhEval disease result with corresponding rank."""
+
+    pheval_disease_result: PhEvalDiseaseResult
+    rank: int
+
+    def as_dict(self):
+        """Return PhEval disease result as dictionary."""
+        return {
+            "disease_name": self.pheval_disease_result.disease_name,
+            "disease_identifier": self.pheval_disease_result.disease_identifier,
+            "score": self.pheval_disease_result.score,
+            "rank": self.rank,
+        }
+
+
 class SortOrder(Enum):
     ASCENDING = 1
     DESCENDING = 2
 
 
 class ResultSorter:
-    def __init__(
-        self, pheval_results: [PhEvalGeneResult] or [PhEvalVariantResult], sort_order: SortOrder
-    ):
+    def __init__(self, pheval_results: Type[PhEvalResult], sort_order: SortOrder):
         self.pheval_results = pheval_results
         self.sort_order = sort_order
 
-    def _sort_by_decreasing_score(self) -> [PhEvalGeneResult] or [PhEvalVariantResult]:
+    def _sort_by_decreasing_score(self) -> Type[PhEvalResult]:
         """Sort results in descending order."""
         return sorted(self.pheval_results, key=operator.attrgetter("score"), reverse=True)
 
-    def _sort_by_increasing_score(self) -> [PhEvalGeneResult] or [PhEvalVariantResult]:
+    def _sort_by_increasing_score(self) -> Type[PhEvalResult]:
         """Sort results in ascending order."""
         return sorted(self.pheval_results, key=operator.attrgetter("score"), reverse=False)
 
-    def sort_pheval_results(self) -> [PhEvalGeneResult] or [PhEvalVariantResult]:
+    def sort_pheval_results(self) -> Type[PhEvalResult]:
```

Comment:
Never seen the `Type[` thing. What does it do? Saying that the result must be a subclass of PhEvalResult?

Reply:
I believe it declares that the result is a class object deriving from PhEvalResult

Reply:
Not so sure.. I would have thought that `Type[MyClass]` does not refer to a concrete subclass of `MyClass`, but rather to the metaclass of `MyClass`. 

Are you sure the return type should not simply be `[PhEvalResult]`?

Reply:
I read into it some more and it looks like you're right!

Reply:
Thanks for checking again, I learned, again, something new :)